src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/BookBinder.java

```diff
@@ -109,23 +115,34 @@ public String getMachineIdentifier() {
 
         for (Map.Entry<Enchantment, Integer> entry : ech2.entrySet()) {
             for (Map.Entry<Enchantment, Integer> conflictsWith : enchantments.entrySet()) {
-                if (entry.getKey().conflictsWith(conflictsWith.getKey())) {
-                    conflicts = true;
+
+                //Check if entry enchantments and conflictsWith enchantment conflict, and confirms that the enchantsments aren't the exact same.
+                if (entry.getKey().conflictsWith(conflictsWith.getKey()) && entry.getKey() != conflictsWith.getKey()) {
+                        conflicts = true;
                 }
```

Comment:
`!=` should never be used to compare objects, only for primitives.

```suggestion
                // Check if entry enchantments and conflictsWith enchantment conflict, and confirms that the enchantsments aren't the exact same.
                if (entry.getKey().conflictsWith(conflictsWith.getKey()) && Objects.equals(entry.getKey(), conflictsWith.getKey())) {
                    conflicts = true;
                }
```